### PR TITLE
Feat/homepageAnnouncement

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

Announcement to ALL users that will see the newer features + folks who are in github login

## Solution

Had to change the API here for business requuirement. (the `onCloseButtonText` is different in this announcement + description is no longer just a text)

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [X] No - this PR is backwards compatible

## Screenshots

**AFTER**:
Screenshot 2023-09-21 at 12 38 06 PM

Screenshot 2023-09-21 at 12 39 19 PM

## Tests

- [ ] Added in storybooks 
- [ ] login and you should see the new announcement (clear local storage / go incognito mode if you dont see it)

### CODE REVIEW

File: src/layouts/Sites.tsx [hunk 1]
```diff
@@ -140,10 +140,10 @@ const SitesContent = ({ siteNames }: { siteNames?: SiteData[] }) => {
 }
 
 export const Sites = (): JSX.Element => {
-  const { email, userId, contactNumber } = useLoginContext()
+  const { email, contactNumber } = useLoginContext()
   const { data: siteRequestData } = useGetAllSites(email)
-  const { announcements, link } = useAnnouncements()
-  const [isOpen, setIsOpen] = useState(announcements.length > 0 && !userId)
+  const { announcements, link, onCloseButtonText } = useAnnouncements()
+  const [isOpen, setIsOpen] = useState(announcements.length > 0)
```
Comment (seaerchin) [inline src/layouts/Sites.tsx:146]: can i ask why we changed the conditionals here?

**Review by seaerchin — COMMENTED** [review]
sanity check on conditional change, rest lgtm

File: src/layouts/Sites.tsx (hunk 1 above)
Comment (kishore03109) [reply src/layouts/Sites.tsx:146]: Sure, previously we gated this to only the newer identity folks. 
At the first render of the `Sites.tsx`, we get `userId = undefined` for email users +  `userId = <userId>` for those who are in gh login. 
This announcement does not follow that semantics, and as such this conditional is removed.

**Comment by kishore03109** [comment]
!run chromatic

**Comment by kishore03109** [comment]
!run chromatic

File: src/features/AnnouncementModal/AnnouncementModal.tsx [hunk 2]
```diff
@@ -58,6 +60,18 @@ export const AnnouncementModal = ({
 
   const { title, description, image, tags } = announcements[currActiveIdx]
 
+  let descComponent: JSX.Element = <></>
+  if (typeof description === "string") {
+    descComponent = (
+      <Text textStyle="body-1" color="base.content.default">
+        {" "}
+        {description}{" "}
```
Comment (dcshzj) [inline src/features/AnnouncementModal/AnnouncementModal.tsx:68]: I think these spaces should be removed. It is causing the description to not be aligned correctly:

Screenshot 2023-09-26 at 10 35 03 AM

Comment (kishore03109) [reply src/features/AnnouncementModal/AnnouncementModal.tsx]: wah good catch 
db12145f288fed970215bb9fef7225d5a6c39c34

**Review by dcshzj — APPROVED** [review]

**Pull request merged**